session/pingpong/exchange_message_tracker.go

```diff
@@ -258,6 +257,46 @@ func (emt *ExchangeMessageTracker) issueExchangeMessage(invoice crypto.Invoice)
 func (emt *ExchangeMessageTracker) Stop() {
 	emt.once.Do(func() {
 		log.Debug().Msg("Stopping...")
+		_ = emt.deps.EventBus.Unsubscribe(connection.AppTopicConsumerStatistics, emt.consumeDataTransferedEvent)
 		close(emt.stop)
 	})
 }
+
+func (emt *ExchangeMessageTracker) consumeDataTransferedEvent(e connection.SessionStatsEvent) {
+	// skip irrelevant sessions
+	if strings.ToLower(string(e.SessionInfo.SessionID)) != strings.ToLower(emt.deps.SessionID) {
+		return
+	}
+
+	// From a server perspective, bytes up are the actual bytes the client downloaded(aka the bytes we pushed to the consumer)
+	// To lessen the confusion, I suggest having the bytes reversed on the session instance.
+	// This way, the session will show that it downloaded the bytes in a manner that is easier to comprehend.
+	emt.updateDataTransfer(e.Stats.BytesSent, e.Stats.BytesReceived)
+}
+
+func (emt *ExchangeMessageTracker) updateDataTransfer(up, down uint64) {
+	emt.dataTransferedLock.Lock()
+	defer emt.dataTransferedLock.Unlock()
+
+	newUp := emt.dataTransfered.up
+	if up > emt.dataTransfered.up {
```

Comment:
Why can't you trust data coming in? If there is bugs, should they be fixes

Reply:
never trust anyone, if data starts decreasing we'll end up either way be notified by the desync of the provider.